datmo/cli/parser.py

```diff
@@ -1,223 +1,219 @@
-    "--config-filepath",
-    dest="config_filepath",
-    default=None,
-    help="absolute filepath to use to search for configuration JSON")
-
-snapshot_create.add_argument(
-    "--stats-filename",
-    dest="stats_filename",
-    default=None,
-    help="filename to use to search for metrics JSON")
-snapshot_create.add_argument(
-    "--stats-filepath",
-    dest="stats_filepath",
-    default=None,
-    help="absolute filepath to use to search for metrics JSON")
-
-snapshot_delete = snapshot_subcommand_parsers.add_parser(
-    "delete", help="Delete a snapshot by id")
-snapshot_delete.add_argument("--id", dest="id", help="snapshot id to delete")
-
-snapshot_ls = snapshot_subcommand_parsers.add_parser(
-    "ls", help="List snapshots")
-snapshot_ls.add_argument(
-    "--session-id",
-    dest="session_id",
-    default=None,
-    help="session id to filter")
-snapshot_ls.add_argument(
-    "--all",
-    "-a",
-    dest="details",
-    action="store_true",
-    help="show detailed snapshot information")
-
-snapshot_checkout = snapshot_subcommand_parsers.add_parser(
-    "checkout", help="checkout a snapshot by id")
-snapshot_checkout.add_argument(
-    "--id", dest="id", default=None, help="snapshot id")
-
-# Task
-task_parser = subparsers.add_parser("task", help="task module")
-task_subcommand_parsers = task_parser.add_subparsers(
-    title="subcommands", dest="subcommand")
-
-# Task run arguments
-task_run = task_subcommand_parsers.add_parser("run", help="run task")
-task_run.add_argument(
-    "--gpu",
-    dest="gpu",
-    action="store_true",
-    help="Boolean if you want to run using GPUs")
-task_run.add_argument(
-    "--ports",
-    "-p",
-    dest="ports",
-    default=None,
-    action="append",
-    type=str,
-    help="""
-    network port mapping during task (e.g. 8888:8888). Left is the host machine port and right
-    is the environment port available during a run.
-""")
-# run.add_argument("--data", nargs="*", dest="data", type=str, help="Path for data to be used during the Task")
-task_run.add_argument(
-    "--environment-def",
-    dest="environment_definition_filepath",
-    default=None,
-    type=str,
-    help=
-    "absolute filepath to environment definition file (e.g. /path/to/Dockerfile)"
-)
-task_run.add_argument(
-    "--interactive",
-    dest="interactive",
-    action="store_true",
-    help="run the environment in interactive mode (keeps STDIN open)")
-task_run.add_argument("cmd", nargs="?", default=None)
-
-# Task list arguments
-task_ls = task_subcommand_parsers.add_parser("ls", help="list tasks")
-task_ls.add_argument(
-    "--session-id",
-    dest="session_id",
-    default=None,
-    nargs="?",
-    type=str,
-    help="pass in the session id to list the tasks in that session")
-
-# Task stop arguments
-task_stop = task_subcommand_parsers.add_parser("stop", help="stop tasks")
-task_stop.add_argument(
-    "--id", dest="id", default=None, type=str, help="task id to stop")
+
```

Comment:
I'll have to test this with the docs code to see if sphinx is able to pick it up. not sure if it will

Reply:
Not sure how this works but meant to reference the entire file and specifically the get_datmo_parser function

Reply:
Looks like this should work

https://sphinx-argparse.readthedocs.io/en/stable/usage.html

```
.. argparse::
   :module: my.module
   :func: my_func_that_returns_a_parser
   :prog: fancytool
```